src/main/java/uk/gov/hmcts/reform/sscs/service/CoverLetterService.java

```diff
@@ -59,14 +74,32 @@ private void printCoverLetterToPdfLocallyForDebuggingPurpose(byte[] coverLetterC
 
         requireNonNull(caseData, "caseData must not be null");
 
-        byte[] coverLetterContent = pdfGenerationService.generatePdf(DocumentHolder.builder()
-            .template(new Template(templateName, hmctsDocName))
-            .placeholders(furtherEvidencePlaceholderService.populatePlaceholders(caseData, letterType))
-            .pdfArchiveMode(true)
-            .build());
+        Map<String, Object> placeholders = furtherEvidencePlaceholderService.populatePlaceholders(caseData, letterType);
+        return generateCoverLetterRetry(letterType, templateName, hmctsDocName, placeholders, 1);
+    }
+
+    public byte[] generateCoverLetterRetry(FurtherEvidenceLetterType letterType, String templateName,
+                                      String hmctsDocName, Map<String, Object> placeholders, int retries) {
+        try {
+
+            byte[] coverLetterContent = pdfGenerationService.generatePdf(DocumentHolder.builder()
+                .template(new Template(templateName, hmctsDocName))
+                .placeholders(placeholders)
+                .pdfArchiveMode(true)
+                .build());
 
+            printCoverLetterToPdfLocallyForDebuggingPurpose(coverLetterContent, letterType, hmctsDocName);
 
-        printCoverLetterToPdfLocallyForDebuggingPurpose(coverLetterContent, letterType, hmctsDocName);
-        return coverLetterContent;
+            return coverLetterContent;
+
+        } catch (Exception e) {
+            if (retries < maxRetryAttempts) {
+                log.info("Retrying Cover Letter Service retry " + retries + " due to " + e.getMessage());
+                return generateCoverLetterRetry(letterType, templateName, hmctsDocName, placeholders, retries + 1);
```

Comment:
Verified locally that this branch is exercised by generateCoverLetterHandleError test

Reply:
code coverage wants a unit test for this line

Reply:
see comment above. generateCoverLetterHandleError exercises this